internal/transport/controlbuf.go

```diff
@@ -990,8 +975,31 @@ func (l *loopyWriter) processData() (bool, error) {
 	if dataItem.onEachWrite != nil {
 		dataItem.onEachWrite()
 	}
-	if err := l.framer.fr.WriteData(dataItem.streamID, endStream, buf[:size]); err != nil {
-		return false, err
+
+	switch fr := l.framer.fr.(type) {
```

Comment:
I need to take another look, but I think loopy could use 'framer' as the GRPCFramer interface and be unaware of the underlying implementation. I have an idea where we could use a Adapter Pattern of sorts to create a adapter to convert from WriteData to WriteDataN.
From the loopy, we should just be able to call WriteData of a slice of []byte.

Something in my mind: processData is in the hot path for the loopyWriter. A loopy is created once per transport. And per this diff, a transport can only speak on type of framer.